src/java/org/apache/cassandra/schema/ColumnMetadata.java

```diff
@@ -556,6 +555,22 @@ public AbstractType<?> getExactTypeIfKnown(String keyspace)
         return type;
     }
 
+    /**
+     * Returns the types of the differents columns.
+     *
+     * @param columns the columns for which the types must be returned.
+     * @return the types of the differents columns.
+     */
+    public static List<AbstractType<?>> typesOf(List<ColumnMetadata> columns)
+    {
+        List<AbstractType<?>> types = new ArrayList<>(columns.size());
+        for (ColumnMetadata column : columns)
+        {
+            types.add(column.type);
+        }
+        return types;
```

Comment:
Would be a sweet on-liner...
```
return columns.stream().map(c -> c.type).collect(Collectors.toList());
```
...but it's probably too bloated, unless the compiler has gotten smarter :D

Reply:
Berenguer does not like streams ;-)

Reply:
Not true. I like them a lot. But I don't like taking risks for no reason...